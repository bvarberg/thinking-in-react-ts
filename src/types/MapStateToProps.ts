import { State } from "./State"

export type MapStateToProps<Props, PropsOuter = {}> = (
  state: State,
  propsOuter: PropsOuter,
) => Props
