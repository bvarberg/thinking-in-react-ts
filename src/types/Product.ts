/**
 * Product - as represented by the starting point API example
 */
export interface Product {
  category: string
  price: string
  stocked: boolean
  name: string
}
