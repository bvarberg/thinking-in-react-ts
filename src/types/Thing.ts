/**
 * Thing - useless example interface
 * ------------------------------------------------
 * Must have height and width properties defined
 */

export interface Thing {
  width: number
  height: number
}
