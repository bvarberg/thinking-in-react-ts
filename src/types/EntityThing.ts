import { Entity } from "./Entity"
import { Thing } from "./Thing"

/**
 * EntityThing - Entity and Thing
 * ------------------------------------------------
 * Must have id, height, width properties defined
 */

export interface EntityThing extends Entity, Thing {}
