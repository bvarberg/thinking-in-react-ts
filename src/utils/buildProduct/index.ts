import { Product } from "../../types/Product"

type BuildProduct = () => Product

export const buildProduct: BuildProduct = () => {
  return {
    category: "Sporting Goods",
    name: "Football",
    price: "$49.99",
    stocked: true,
  }
}
