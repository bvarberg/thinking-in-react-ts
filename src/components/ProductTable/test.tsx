import { ProductTable } from "."
import React from "react"
import { shallowRender } from "../../utils/shallowRender"
import { Product } from "../../types/Product"

describe("ProductTable", () => {
  it("renders", () => {
    const products: Product[] = []

    const received = shallowRender(<ProductTable products={products} />)

    expect(received).toMatchSnapshot()
  })

  describe("with a few products", () => {
    const products: Product[] = [
      {
        category: "Sporting Goods",
        price: "$49.99",
        stocked: true,
        name: "Football",
      },
      {
        category: "Sporting Goods",
        price: "$9.99",
        stocked: true,
        name: "Baseball",
      },
      {
        category: "Sporting Goods",
        price: "$29.99",
        stocked: false,
        name: "Basketball",
      },
      {
        category: "Electronics",
        price: "$99.99",
        stocked: true,
        name: "iPod Touch",
      },
      {
        category: "Electronics",
        price: "$399.99",
        stocked: false,
        name: "iPhone 5",
      },
    ]

    const received = shallowRender(<ProductTable products={products} />)

    expect(received).toMatchSnapshot()
  })
})
