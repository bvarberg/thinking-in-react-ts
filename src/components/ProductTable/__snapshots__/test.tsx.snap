// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<table
  className="ProductTable"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Price
      </th>
    </tr>
  </thead>
  <tbody>
    <ProductCategoryRow
      category="Sporting Goods"
    />
    <ProductRow
      product={
        Object {
          "category": "Sporting Goods",
          "name": "Football",
          "price": "$49.99",
          "stocked": true,
        }
      }
    />
    <ProductRow
      product={
        Object {
          "category": "Sporting Goods",
          "name": "Baseball",
          "price": "$9.99",
          "stocked": true,
        }
      }
    />
    <ProductRow
      product={
        Object {
          "category": "Sporting Goods",
          "name": "Basketball",
          "price": "$29.99",
          "stocked": false,
        }
      }
    />
    <ProductCategoryRow
      category="Electronics"
    />
    <ProductRow
      product={
        Object {
          "category": "Electronics",
          "name": "iPod Touch",
          "price": "$99.99",
          "stocked": true,
        }
      }
    />
    <ProductRow
      product={
        Object {
          "category": "Electronics",
          "name": "iPhone 5",
          "price": "$399.99",
          "stocked": false,
        }
      }
    />
  </tbody>
</table>
`;

exports[`ProductTable renders 1`] = `
<table
  className="ProductTable"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Price
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
